iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/storable_function.hpp

```diff
@@ -112,12 +112,12 @@ class storable_function<StorageType, signature<ReturnType, Args...>>
     ///          Not specifying move or using a default implementation is fine.
     ///          This is also the case for std::function (for the gcc implementation at least).
     ///
-    ReturnType operator()(Args... args);
+    ReturnType operator()(Args... args) const;
```

Comment:
@MatthiasKillat maybe we need to add a test for the scenario you described. If that does not break, then I see no reason to revert the change

Reply:
`test_cxx_function.cpp` provides a test class `Functor` which state is changed when `operator()` is called and all the corresponding tests pass.

Reply:
@elboberido The test exists and does not break because the memory that is modified by the functor (and owned by function) is type erased, so the compiler cannot know/prove that it is modified (or not modified).

I think the usage problem arises if we have a `const cxx::function` and want to call `operator()`. To resolve this, we can make `operator()` `const` but this is mainly a workaround. I see no easy way to get proper `const overloading` where the methods are actually different.

The reason for this is basically that at construction time it needs to be checked whether it is a functor with state (meaning its `operator()` cannot be const. For these the wrapping `cxx::function` would need to allow only non-const `operator()`.

TLDR: this may need complex case distinctions in `cxx::function` to really get right (if this is even possible). If for now it is sufficient and necessary to make it `const` we can do so.